docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,668 @@
+    ```sql
+    SELECT aer.*
+    FROM alarm_event_record aer
+             JOIN alarm_subscription_info asi ON asi.alarm_subscription_id = 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11'
+    WHERE aer.alarm_sequence_number > asi.largest_number_alarm_event_seen_so_far
+      and aer.perceived_severity = 'CRITICAL'
+    ORDER BY aer.alarm_sequence_number;
+    ```
+- Process and notify by deriving `AlarmEventRecordModifications` o-ran DS + callback.
+- Update sequence for subscription indicating the latest event sent so far
+    ```go
+    var largestProcessedSequenceNumber int64
+    // for each alarms Update the largest sequence number we've processed
+    largestProcessedSequenceNumber = max(alarm.SequenceNumber, largestProcessedSequenceNumber))
+    // The last row in the query is always the largest (ORDER BY default is accending)
+    ```
+  And finally update 
+  ```sql
+  UPDATE alarm_subscription_info
+  SET largest_number_alarm_event_seen_so_far = $largestProcessedSequenceNumber
+  WHERE alarm_subscription_id = 'a2eebc99-9c0b-4ef8-bb6d-6bb9bd380a13';
+  ```
+- NOTE: `alarm_sequence_number` is automatically handled from inside the DB. When the sequence increments, subscriber is notified. 
+  See notification conditions [here](#conditions-for-notifying-subscriber)
+  
+### Conditions for Notifying subscriber
+- When alarm is `firing` for the first time
+- When alarm is updated to `resolved`
+- When `alarm_changed_time` changes (could be multiple times)
+
+### Daily archive cleanup
+Run this using a k8s CronJob CR at the start of every hour
+
+```sql
+DELETE FROM alarm_event_record_archive
+WHERE alarm_cleared_time < NOW() - INTERVAL '24 hour' and status = 'resolved';
+```
+We can apply the CR before server starts and remove it during shutdown as part of teardown e.g inside `server.RegisterOnShutdown`
+
+## K8s resources
+We will need few K8s resources that will be eventually applied by the Operator. 
+
+#### Alarm server
+This is essentially a typical CRUD app and we need the following
+
+- Deployment: It should have one initContainers that runs to completion to perform DB migration and one main container which starts the main server. 
+  No HA, so set replica to 1. It should also contain all the ENV variables needed to talk to postgres deployment DB_HOST, DB_PORT, DB_NAME, DB_USER etc.
+  Suitable resources should be provided but not much memory and CPU is need for CRUD apps. 
+- Secrets and ConfigMap: DB creds and configs should be read from here. We can probably use the configs from Postgres (seen below) for user and password. 
+- Service: Expose and balance using `ClusterIP` (though to start with we will set replica to 1)
+- Ingress: Expose service so that it can be called by users from outside the cluster
+
+
+#### Postgres
+This deployment can be leveraged by many microservices by creating their own Database. 
```

Comment:
This needs more detail.

1. How are the credentials being exposed and consumed for the various servers in the o-cloud operator that need a DB
2. There is a requirement that you have replicated persistent storage 
3. How do the clients discover the ClusterIP ?
4. How are you configuring the DB, where does this configuration come from ?

Reply:
Added all the info, including default username and password, and where it can be stored. Also an example connection string that an can use which includes how client can discover the postgres service

Also specified that ODF is required in the hub